### PR TITLE
Expand chatinterface to full window height

### PR DESCRIPTION
**Pull request by aliabid94**
ChatInterfaces (and other Blocks optionally) take full screen height. Rearranged interface a little:

Recording 2024-02-01 at 16 48 17

Closes: https://github.com/gradio-app/gradio/issues/4001

### CODE REVIEW

**Comment by aliabid94** [comment]
PR #7278 got in a weird git state so trying again.

File: gradio/chat_interface.py [hunk 1]
```diff
@@ -170,85 +173,84 @@ def __init__(
             if description:
                 Markdown(description)
```
Comment (aliabid94) [inline gradio/chat_interface.py:175]: looks like a lot of change but just moved the chat buttons before the textbox and got rid of the parent panel column

File: gradio/blocks.py [hunk 2]
```diff
@@ -514,6 +515,7 @@ def __init__(
             css: Custom css as a string or path to a css file. This css will be included in the demo webpage.
             js: Custom js or path to js file to run when demo is first loaded. This javascript will be included in the demo webpage.
             head: Custom html to insert into the head of the demo webpage. This can be used to add custom meta tags, scripts, stylesheets, etc. to the page.
+            expand_to_height: Whether to vertically expand top-level child components to the height of the window. If True, expansion occurs when the scale value of the child components >= 1.
```
Comment (abidlabs) [inline gradio/blocks.py:518]: This means that the Gradio app will be the full height of the window, right? I think `full_height` would be a better parameter name

Comment (abidlabs) [reply gradio/blocks.py:518]: Should we add this parameter to `Interface` and `TabbedInterface` too btw?

Comment (abidlabs) [reply gradio/blocks.py:518]: So just to understand, the same `scale` parameter is used to determine how much the height of each child component should expand by? If so, we should update the docstring for `scale` for each component to mention this. If not, how does this work if you have multiple child elements whose height can vary?

**Comment by abidlabs** [comment]
@aliabid94 the full-screen `gr.ChatInterface` looks great but I'm confused about how to use the `expand_to_height` parameter more generally. For example, neither of these work:

```py
import gradio as gr

with gr.Blocks(expand_to_height=True) as demo:
    gr.Chatbot(scale=2)
    gr.Chatbot(scale=2)
    
demo.launch()
```

(This^ takes up more than the full height of the page)

Nor this:

```py
import gradio as gr

with gr.Blocks(expand_to_height=True) as demo:
    gr.Textbox(scale=2)
    gr.Textbox(scale=2)
    
demo.launch()
```

(This^ takes up less than the full height)

Some additional explanation of this parameter in the Guides would be useful imo.

File: gradio/blocks.py (hunk 2 above)
Comment (aliabid94) [reply gradio/blocks.py]: >  I think full_height or fill_height would be a better parameter name

Renamed to `fill_height`. 

> Should we add this parameter to Interface and TabbedInterface too btw?

I don't think it's necessary. 

> If so, we should update the docstring for scale for each component to mention this.

Ok updated docstring of every component.

**Comment by aliabid94** [comment]
> For example, neither of these work:
>
> ```
> import gradio as gr
> 
> with gr.Blocks(expand_to_height=True) as demo:
>     gr.Chatbot(scale=2)
>     gr.Chatbot(scale=2)
>     
> demo.launch()
> ```
> 
>
> (This^ takes up more than the full height of the page)

Chatbots by default have a minimum height of 400 if no height is specified. If an app extends past the window, fill_height does not shrink any element, the height of the elemnts are respected first. To "fix" this, simply set the height to something smaller, and the Components will expand.


> Nor this:
> 
> ```
> import gradio as gr
> 
> with gr.Blocks(expand_to_height=True) as demo:
>     gr.Textbox(scale=2)
>     gr.Textbox(scale=2)
>     
> demo.launch()
> (This^ takes up less than the full height)
> ```

Fixed. Was an issue with the autogenerated Form component not respecting scale.

**Comment by abidlabs** [comment]
Very nice @aliabid94! LGTM

**Review by abidlabs — APPROVED** [review]

**Comment by abidlabs** [comment]
Merging so that we can do a release

**Comment by kostinmax** [comment]
Could you please provide a working example of how to use this (fill_height) new property?

**Pull request merged**